### ISSUE TITLE
Upcoming version of `rredlist` breaks `taxize`

### ISSUE THREAD

**Issue by willgearty** [ropensci/taxize#937]
I'm the maintainer of the `rredlist` package, and I wanted to make you aware that the next version of the `rredlist` package will break the IUCN functions in the `taxize` package. This is due to major changes in the IUCN API (v4, see discussion [here](https://github.com/ropensci/rredlist/issues/52)), which I'm planning to track with the next version of `rredlist`. I'm removing all functionality related to the old API version (v3) from the package, which several functions in `taxize` depend on, but I'd be happy to work with you all to migrate some (or all) of the needed functionality over to `taxize`. Let me know how I can help.

**Comment by dcuadrac** [comment]
Thank you so much!

**Comment by zachary-foster** [comment]
Thanks for the info! I am working on a new CRAN release for taxize, so I will try to work these changes in. I will take a loko at the  `api_v4` branch, which I assume is where the relevant changes are?

**Comment by willgearty** [comment]
Yep, that's the one!

**Comment by zachary-foster** [comment]
Thanks! It looks like there is no search for taxa in general now? I see functions for specific taxonomic levels (e.g. `rl_species`) but no generic search. This will make it difficult to make the IUCN functions in `taxize` work the same as functions for other data sources, since I dont see a way to know which taxonomic rank a random character supplied by a user is.

**Comment by willgearty** [comment]
That is correct, there is no longer any general taxonomic search in the IUCN API, see all of the endpoint options here: https://api.iucnredlist.org/api-docs/index.html.

All of the IUCN examples in the `taxize` documentation use <Genus species>, and half of the functions are even explicit in the docs that this is the format (https://docs.ropensci.org/taxize/reference/iucn_summary.html, https://docs.ropensci.org/taxize/reference/iucn_id.html). Are the other two functions also supposed to work for other taxonomic levels?

**Comment by willgearty** [comment]
FYI, the changes have been merged to the main branch, so [r-universe](https://ropensci.r-universe.dev/rredlist) and the package's [website](https://docs.ropensci.org/rredlist/) are now updated to new version of the package. I'm hoping to publish to CRAN as soon as possible, so let me know if you have any other questions.

**Comment by zachary-foster** [comment]
> Are the other two functions also supposed to work for other taxonomic levels?

I assumed so since that's how the other data source functions, but perhaps it was not an option for this one. 

I am trying to get  fix all the issues that have accumulated for `taxize` but I dont have a lot of free time for it. Do you think there is value in having versions of the IUCN functions in `taxize` or would just redirecting users to `rredlist` be better?

**Comment by willgearty** [comment]
As far as I can tell, `taxize` calls the following `rredlist` functions:

- `rl_use_iucn()`: used to set up an API key, no endpoints called
- `rl_search()` (admittedly a misleading function name): used the old [/species/ endpoint](https://apiv3.iucnredlist.org/api/v3/docs#species-individual-name)
- `rl_common_names()`: used the old [/species/common_names/ endpoint](https://apiv3.iucnredlist.org/api/v3/docs#species-common)
- `rl_synonyms()`: used the old [/species/synonyms/ endpoint](https://apiv3.iucnredlist.org/api/v3/docs#species-synonym)
- `rl_history()`: used the old [/species/history/name endpoint](https://apiv3.iucnredlist.org/api/v3/docs#species-history-name)
- `rl_occ_country()`: used the old [/species/countries/name endpoint](https://apiv3.iucnredlist.org/api/v3/docs#species-occurrence-name)

All of these endpoints only support binomials, so it appears to me that these `taxize` functions only ever supported binomials as well?

Therefore, I don't think there should be too much work to do

**Comment by willgearty** [comment]
@zachary-foster [based on feedback by some beta testers](https://github.com/ropensci/rredlist/pull/57), I added any old functions that I had removed back in as defunct functions and added lifecycle errors if they are ever used. So this update should no longer cause `taxize` to break, but it will still result in errors for users if they try to use `taxize` functions that call any of these now-defunct errors (and may break `taxize` tests).

At the moment, `rredlist` is going through [rOpenSci peer review](https://github.com/ropensci/software-review/issues/663), so it'll be at least a few weeks (if not more than a month) before that wraps up and the new version heads to CRAN. Let me know if you need anymore information or insight about `rredlist` as you work towards the next version of `taxize`.

**Comment by zachary-foster** [comment]
I have updated the taxize functions to use the new functions in `rredlist` (e.g. `rl_species`). I am about ready to release the updated `taxize` back to CRAN. Looks like `rredlist` just went through ropensci peer review. Any estimate on when the CRAN update will happen? Thanks!

**Comment by willgearty** [comment]
The new `rredlist` is on CRAN as of this afternoon 👍

**Comment by zachary-foster** [comment]
Wow, good timing! Thanks!
